main.tf

```diff
@@ -1,81 +1,205 @@
-# Set our cloud provider and access details
+################################################################################
+# Set our cloud provider and access details,
+# such as AWS security credentials
+# https://console.aws.amazon.com/iam/home?region=us-east-1#/security_credentials
+# it stored in ./terraform.tdvars
+#
+################################################################################
 provider "aws" {
   access_key = var.access_key
   secret_key = var.secret_key
   region     = var.region
 }
 
-# Create a VPC to launch our instances into
+data "aws_availability_zones" "available" {}
+################################################################################
+# Create a VPC to launch main VPC-network
+# determine the CIDR block for our VPC
+#
+#
+# https://www.terraform.io/docs/providers/aws/d/vpc.html
+################################################################################
 resource "aws_vpc" "main-vpc" {
-  cidr_block        = "172.20.0.0/16"
-  security_group_id = "${aws_security_group.front.id}"
+  cidr_block           = var.cidr["main"]
+  enable_dns_hostnames = true
+  enable_dns_support   = "true"
   tags = {
-    Name = "VI_PI_SI"
+    Name = "MAIN_VPC"
   }
 }
-
-# Define subnet for back-services
-resource "aws_subnet" "main-subnet" {
-  vpc_id     = "${aws_vpc.main-vpc.id}"
-  cidr_block = "172.20.0.0/24"
-
-  tags = {
-    Name = "SUBNET_back-services"
+################################################################################
+# Provides a resource to create a VPC Internet Gateway
+# vpc_id - (Required) The VPC ID to create in
+#
+# https://www.terraform.io/docs/providers/aws/r/internet_gateway.html
+################################################################################
+resource "aws_internet_gateway" "gw" {
+  vpc_id = "${aws_vpc.main-vpc.id}"
```

Comment:
Since terraform 0.12 you don't need to use "${ }" syntax for value defining (in case when no interpolation needed)
So, it might be just aws_vpc.main-vpc.id

Reply:
ok, thnx)